specs/merge/beacon-chain.md

```diff
@@ -209,41 +200,37 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
     process_randao(state, block.body)
     process_eth1_data(state, block.body)
     process_operations(state, block.body)
-    # Pre-merge, skip execution payload processing
-    if is_execution_enabled(state, block):
-        process_execution_payload(state, block.body.execution_payload, EXECUTION_ENGINE)  # [New in Merge]
+    if is_execution_enabled(state, block.body):
+        process_execution_payload(state, block.body, EXECUTION_ENGINE)  # [New in Merge]
 ```
 
-#### Execution payload processing
+### Execution payload processing
 
-##### `process_execution_payload`
+#### `process_execution_payload`
 
 ```python
-def process_execution_payload(state: BeaconState,
-                              execution_payload: ExecutionPayload,
-                              execution_engine: ExecutionEngine) -> None:
-    """
-    Note: This function is designed to be able to be run in parallel with the other `process_block` sub-functions
-    """
-    if is_transition_completed(state):
-        assert execution_payload.parent_hash == state.latest_execution_payload_header.block_hash
-        assert execution_payload.number == state.latest_execution_payload_header.number + 1
-
-    assert execution_payload.timestamp == compute_time_at_slot(state, state.slot)
-
-    assert execution_engine.new_block(execution_payload)
-
+def process_execution_payload(state: BeaconState, body: BeaconBlockBody, execution_engine: ExecutionEngine) -> None:
```

Comment:
Changing `execution_payload` to `body` in the function signature requires corresponding change in execution payload tests, they will have to use beacon blocks instead of pure payloads. Though, it might not be undesirable, just requires additional work.

Reply:
> just requires additional work

Right. I'm happy to do the work to fix tests :)

Reply:
I'm really not a fan of the `execution_payload -> body` change, it completely breaks the encapsulation of the processing. Can we just keep it `def process_execution_payload(state: BeaconState, payload: ExecutionPayload, engine: ExecutionEngine) -> None:` ?

Reply:
now passing the payload to `process_execution_payload` :)